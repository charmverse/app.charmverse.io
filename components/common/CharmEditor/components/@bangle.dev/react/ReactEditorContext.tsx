import type { EditorView } from 'prosemirror-view';
import type { Dispatch, ReactNode, SetStateAction } from 'react';
import { createContext, useContext, useMemo, useState } from 'react';

// keep track of the focused page (may be different from what's in the URL or header)
type ICharmEditorView = {
  view: EditorView | null;
  setView: (view: EditorView) => void;
};

const CharmEditorView = createContext<Readonly<ICharmEditorView>>({
  view: null,
  setView: () => {}
});

export function CharmEditorViewProvider({ children }: { children: ReactNode }) {
  const [view, setView] = useState<EditorView | null>(null);

  const value: ICharmEditorView = useMemo(
    () => ({
      view,
      setView
    }),
    [view, setView]
  );

  return <CharmEditorView.Provider value={value}>{children}</CharmEditorView.Provider>;
}

export const useCharmEditorView = () => useContext(CharmEditorView);
