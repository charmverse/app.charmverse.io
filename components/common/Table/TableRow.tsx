import TableRow from '@mui/material/TableRow';
import styled from '@emotion/styled';

const StyledRow = styled(TableRow)`
  .row-actions {
    opacity: 0;
    transition: opacity 0.2s ease-in-out;
  }
  &:hover .row-actions {
    opacity: 1;
  }
`;

export default StyledRow;
