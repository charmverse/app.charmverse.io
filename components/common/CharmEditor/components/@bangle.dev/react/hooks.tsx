import type { EditorView } from '@bangle.dev/pm';
import { Plugin, PluginKey } from '@bangle.dev/pm';
import { rafSchedule } from '@bangle.dev/utils';
import { useContext, useEffect, useState } from 'react';

import type { BangleEditorStateProps } from 'components/common/CharmEditor/components/@bangle.dev/core/bangle-editor-state';

import { BangleEditorState } from '../core/bangle-editor-state';

import { EditorViewContext } from './ReactEditor';

export function useEditorState(props: BangleEditorStateProps) {
  if (props.plugins && typeof props.plugins !== 'function') {
    throw new Error('plugins error: plugins must be a function');
  }

  const [state] = useState(
    () =>
      // Instantiate the editorState once and keep using that instance
      // on subsequent renders.
      // Passing a callback in useState lazy calls the
      // functions on the first render and never again.
      new BangleEditorState(props)
  );

  return state;
}

export function usePluginState(pluginKey: PluginKey, throttle = false) {
  const view = useEditorViewContext();
  const [state, setState] = useState(pluginKey.getState(view.state));

  useEffect(() => {
    let _setState = setState;
    if (throttle) {
      _setState = rafSchedule(setState);
    }
    const plugin = watcherPlugin(pluginKey, _setState);
    (view as any)._updatePluginWatcher(plugin);
    return () => {
      if (throttle) {
        (_setState as ReturnType<typeof rafSchedule>).cancel();
      }
      (view as any)._updatePluginWatcher(plugin, true);
    };
  }, [view, pluginKey, throttle]);

  return state;
}

export function useEditorViewContext(): EditorView {
  const context = useContext(EditorViewContext);
  if (!context) {
    throw new Error('useEditorViewContext must be used within a EditorViewContext');
  }
  return context;
}

function watcherPlugin(pluginKey: PluginKey, setState: (state: any) => void) {
  return new Plugin({
    key: new PluginKey(`withPluginState_${(pluginKey as any).key}`),
    view() {
      return {
        update(view, prevState) {
          const { state } = view;
          if (prevState === state) {
            return;
          }
          const newPluginState = pluginKey.getState(state);

          if (newPluginState !== pluginKey.getState(prevState)) {
            setState(newPluginState);
          }
        }
      };
    }
  });
}
