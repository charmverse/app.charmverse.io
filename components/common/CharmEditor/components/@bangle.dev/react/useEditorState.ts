import { useState } from 'react';

import type { BangleEditorStateProps } from '../core/bangle-editor-state';
import { BangleEditorState } from '../core/bangle-editor-state';

export function useEditorState(props: BangleEditorStateProps) {
  if (props.plugins && typeof props.plugins !== 'function') {
    throw new Error('plugins error: plugins must be a function');
  }
  const [state] = useState(
    () =>
      // Instantiate the editorState once and keep using that instance
      // on subsequent renders.
      // Passing a callback in useState lazy calls the
      // functions on the first render and never again.
      new BangleEditorState(props)
  );
  return state;
}
