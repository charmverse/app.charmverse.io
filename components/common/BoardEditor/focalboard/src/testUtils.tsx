import type { PageMeta } from '@charmverse/core/pages';
import React from 'react';
import { DndProvider } from 'react-dnd';
import { HTML5Backend } from 'react-dnd-html5-backend';
import { IntlProvider } from 'react-intl';
import type { Middleware } from 'redux';
import type { MockStoreEnhanced } from 'redux-mock-store';
import configureStore from 'redux-mock-store';
import { v4 } from 'uuid';

import { PagesContext } from 'hooks/usePages';
import type { Block } from 'lib/focalboard/block';
import { pageStubToCreate } from 'testing/generatePageStub';
import { AppThemeProvider } from 'theme/AppThemeProvider';

export const wrapIntl = (children?: React.ReactNode): JSX.Element => (
  <AppThemeProvider>
    <IntlProvider locale='en'>{children}</IntlProvider>
  </AppThemeProvider>
);
export const wrapDNDIntl = (children?: React.ReactNode): JSX.Element => {
  return <DndProvider backend={HTML5Backend}>{wrapIntl(children)}</DndProvider>;
};

export const wrapPagesProvider = (cardId: string | string[], children: React.ReactNode) => {
  const pages: Record<string, PageMeta> = {};

  const cardIds = Array.isArray(cardId) ? cardId : [cardId];

  cardIds.forEach((_cardId) => {
    pages[_cardId] = pageStubToCreate({ spaceId: v4(), createdBy: v4(), id: _cardId, path: _cardId }) as PageMeta;
  });

  return (
    <PagesContext.Provider
      value={
        {
          pages
        } as any
      }
    >
      {children}
    </PagesContext.Provider>
  );
};

export function mockDOM(): void {
  window.focus = jest.fn();
  document.createRange = () => {
    const range = new Range();
    range.getBoundingClientRect = jest.fn();
    range.getClientRects = () => {
      return {
        item: () => null,
        length: 0,
        [Symbol.iterator]: jest.fn()
      };
    };
    return range;
  };
}
export function mockMatchMedia(result: { matches: boolean }): void {
  // We check if system preference is dark or light theme.
  // This is required to provide it's definition since
  // window.matchMedia doesn't exist in Jest.
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(() => {
      return result;

      // return ({
      //     matches: true,
      // })
    })
  });
}

export function mockStateStore(middleware: Middleware[], state: unknown): MockStoreEnhanced<unknown, unknown> {
  const mockStore = configureStore(middleware);
  return mockStore(state);
}

export type BlocksById<BlockType> = { [key: string]: BlockType };

export function blocksById<BlockType extends Block>(blocks: BlockType[]): BlocksById<BlockType> {
  return blocks.reduce((res, block) => {
    res[block.id] = block;
    return res;
  }, {} as BlocksById<BlockType>);
}
