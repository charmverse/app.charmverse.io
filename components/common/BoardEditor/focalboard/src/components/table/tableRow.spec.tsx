
import '@testing-library/jest-dom';
import 'isomorphic-fetch';
import { render } from '@testing-library/react';
import React from 'react';
import { Provider as ReduxProvider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { TestBlockFactory } from '../../test/testBlockFactory';
import { wrapDNDIntl } from '../../testUtils';

import TableRow from './tableRow';

describe('components/table/TableRow', () => {
  const board = TestBlockFactory.createBoard();
  const view = TestBlockFactory.createBoardView(board);

  const view2 = TestBlockFactory.createBoardView(board);
  view2.fields.sortOptions = [];

  const card = TestBlockFactory.createCard(board);
  const cardTemplate = TestBlockFactory.createCard(board);
  cardTemplate.fields.isTemplate = true;

  const state = {
    users: {},
    comments: {
      comments: {}
    },
    contents: {
      contents: {}
    },
    cards: {
      cards: {
        [card.id]: card
      }
    }
  };

  const mockStore = configureStore([]);

  test('should match snapshot', async () => {
    const store = mockStore(state);
    const component = wrapDNDIntl(
      <ReduxProvider store={store}>
        <TableRow
          board={board}
          activeView={view}
          card={card}
          isSelected={false}
          focusOnMount={false}
          onSaveWithEnter={jest.fn()}
          showCard={jest.fn()}
          readOnly={false}
          offset={0}
          resizingColumn=''
          columnRefs={new Map()}
          onDrop={jest.fn()}
        />
      </ReduxProvider>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot, read-only', async () => {
    const store = mockStore(state);
    const component = wrapDNDIntl(
      <ReduxProvider store={store}>
        <TableRow
          board={board}
          card={card}
          activeView={view}
          isSelected={false}
          focusOnMount={false}
          onSaveWithEnter={jest.fn()}
          showCard={jest.fn()}
          readOnly={true}
          offset={0}
          resizingColumn=''
          columnRefs={new Map()}
          onDrop={jest.fn()}
        />
      </ReduxProvider>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot, isSelected', async () => {
    const store = mockStore(state);
    const component = wrapDNDIntl(
      <ReduxProvider store={store}>
        <TableRow
          board={board}
          card={card}
          activeView={view}
          isSelected={true}
          focusOnMount={false}
          onSaveWithEnter={jest.fn()}
          showCard={jest.fn()}
          readOnly={false}
          offset={0}
          resizingColumn=''
          columnRefs={new Map()}
          onDrop={jest.fn()}
        />
      </ReduxProvider>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot, collapsed tree', async () => {
    view.fields.collapsedOptionIds = ['value1'];
    view.fields.hiddenOptionIds = [];

    const store = mockStore(state);
    const component = wrapDNDIntl(
      <ReduxProvider store={store}>
        <TableRow
          board={board}
          card={card}
          activeView={view}
          isSelected={false}
          focusOnMount={false}
          onSaveWithEnter={jest.fn()}
          showCard={jest.fn()}
          readOnly={false}
          offset={0}
          resizingColumn=''
          columnRefs={new Map()}
          onDrop={jest.fn()}
        />
      </ReduxProvider>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot, display properties', async () => {
    view.fields.visiblePropertyIds = ['property1', 'property2'];

    const store = mockStore(state);
    const component = wrapDNDIntl(
      <ReduxProvider store={store}>
        <TableRow
          board={board}
          card={card}
          activeView={view}
          isSelected={false}
          focusOnMount={false}
          onSaveWithEnter={jest.fn()}
          showCard={jest.fn()}
          readOnly={false}
          offset={0}
          resizingColumn=''
          columnRefs={new Map()}
          onDrop={jest.fn()}
        />
      </ReduxProvider>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });

  test('should match snapshot, resizing column', async () => {
    view.fields.visiblePropertyIds = ['property1', 'property2'];

    const store = mockStore(state);
    const component = wrapDNDIntl(
      <ReduxProvider store={store}>
        <TableRow
          board={board}
          card={card}
          activeView={view}
          isSelected={false}
          focusOnMount={false}
          onSaveWithEnter={jest.fn()}
          showCard={jest.fn()}
          readOnly={false}
          offset={0}
          resizingColumn='property1'
          columnRefs={new Map()}
          onDrop={jest.fn()}
        />
      </ReduxProvider>
    );
    const { container } = render(component);
    expect(container).toMatchSnapshot();
  });
});
