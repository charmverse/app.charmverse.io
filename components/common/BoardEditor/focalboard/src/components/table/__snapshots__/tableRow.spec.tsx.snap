// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/table/TableRow should match snapshot 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="TableRow octo-table-row"
      data-test="database-row-card-1"
      style="background-color: transparent; opacity: 1; cursor: move;"
    >
      <div
        class="MuiStack-root css-1v111ru-MuiStack-root"
      >
        <div
          class="icons row-actions"
          draggable="true"
          style="padding: 0px;"
        >
          <div
            class="charm-drag-handle disable-drag-selection MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-d1t9d3-MuiSvgIcon-root"
              data-testid="DragIndicatorIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2m-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell title-cell MuiBox-root css-v7v99c"
      >
        <div
          style="display: flex; width: 100%;"
        >
          <div
            class="octo-icontitle"
            style="align-self: flex-start; align-items: flex-start;"
          >
            <div
              class="css-jm1ma0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hvmj5c-MuiSvgIcon-root"
                data-testid="InsertDriveFileOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zM6 20V4h7v5h5v11z"
                />
              </svg>
            </div>
            <div
              style="width: 100%; height: 100%;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-2umbt5-MuiInputBase-root"
                  spellcheck="true"
                >
                  <input
                    class="MuiInputBase-input Editable octo-propertyvalue css-yz9k0d-MuiInputBase-input"
                    placeholder="Untitled"
                    type="text"
                    value="Page Title"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="open-button"
            data-test="database-open-button-card-1"
          >
            <button
              class="Button size--small"
              type="button"
            >
              <span>
                Open
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="flex-grow: 1;"
      />
    </div>
  </span>
</div>
`;

exports[`components/table/TableRow should match snapshot, collapsed tree 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="TableRow octo-table-row hidden"
      data-test="database-row-card-1"
      style="background-color: transparent; opacity: 1; cursor: move;"
    >
      <div
        class="MuiStack-root css-1v111ru-MuiStack-root"
      >
        <div
          class="icons row-actions"
          draggable="true"
          style="padding: 0px;"
        >
          <div
            class="charm-drag-handle disable-drag-selection MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-d1t9d3-MuiSvgIcon-root"
              data-testid="DragIndicatorIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2m-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell title-cell MuiBox-root css-v7v99c"
      >
        <div
          style="display: flex; width: 100%;"
        >
          <div
            class="octo-icontitle"
            style="align-self: flex-start; align-items: flex-start;"
          >
            <div
              class="css-jm1ma0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hvmj5c-MuiSvgIcon-root"
                data-testid="InsertDriveFileOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zM6 20V4h7v5h5v11z"
                />
              </svg>
            </div>
            <div
              style="width: 100%; height: 100%;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-2umbt5-MuiInputBase-root"
                  spellcheck="true"
                >
                  <input
                    class="MuiInputBase-input Editable octo-propertyvalue css-yz9k0d-MuiInputBase-input"
                    placeholder="Untitled"
                    type="text"
                    value="Page Title"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="open-button"
            data-test="database-open-button-card-1"
          >
            <button
              class="Button size--small"
              type="button"
            >
              <span>
                Open
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="flex-grow: 1;"
      />
    </div>
  </span>
</div>
`;

exports[`components/table/TableRow should match snapshot, display properties 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="TableRow octo-table-row hidden"
      data-test="database-row-card-1"
      style="background-color: transparent; opacity: 1; cursor: move;"
    >
      <div
        class="MuiStack-root css-1v111ru-MuiStack-root"
      >
        <div
          class="icons row-actions"
          draggable="true"
          style="padding: 0px;"
        >
          <div
            class="charm-drag-handle disable-drag-selection MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-d1t9d3-MuiSvgIcon-root"
              data-testid="DragIndicatorIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2m-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell title-cell MuiBox-root css-v7v99c"
      >
        <div
          style="display: flex; width: 100%;"
        >
          <div
            class="octo-icontitle"
            style="align-self: flex-start; align-items: flex-start;"
          >
            <div
              class="css-jm1ma0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hvmj5c-MuiSvgIcon-root"
                data-testid="InsertDriveFileOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zM6 20V4h7v5h5v11z"
                />
              </svg>
            </div>
            <div
              style="width: 100%; height: 100%;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-2umbt5-MuiInputBase-root"
                  spellcheck="true"
                >
                  <input
                    class="MuiInputBase-input Editable octo-propertyvalue css-yz9k0d-MuiInputBase-input"
                    placeholder="Untitled"
                    type="text"
                    value="Page Title"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="open-button"
            data-test="database-open-button-card-1"
          >
            <button
              class="Button size--small"
              type="button"
            >
              <span>
                Open
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="align-items: flex-start; width: 100px; overflow-x: hidden;"
      >
        <div
          style="width: 100%; height: 100%;"
        >
          <div
            style="height: 100%;"
          >
            <div
              class="MuiStack-root css-1gl94ry-MuiStack-root"
              data-test="closed-select-input"
            >
              <div
                class="MuiStack-root css-1a59yxd-MuiStack-root"
                data-test="select-preview"
              >
                <div
                  class="MuiStack-root css-b3ouwx-MuiStack-root"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorTurquoise MuiChip-filledTurquoise css-zuoxmo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                    >
                      value 1
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="align-items: flex-start; width: 100px; overflow-x: hidden;"
      >
        <div
          style="width: 100%; height: 100%;"
        >
          <div
            style="height: 100%;"
          >
            <div
              class="MuiStack-root css-1gl94ry-MuiStack-root"
              data-test="closed-select-input"
            >
              <div
                class="MuiStack-root css-1a59yxd-MuiStack-root"
                data-test="select-preview"
              >
                <div
                  class="MuiStack-root css-b3ouwx-MuiStack-root"
                  data-mui-internal-clone-element="true"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="flex-grow: 1;"
      />
    </div>
  </span>
</div>
`;

exports[`components/table/TableRow should match snapshot, isSelected 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="TableRow octo-table-row selected"
      data-test="database-row-card-1"
      style="background-color: transparent; opacity: 1; cursor: move;"
    >
      <div
        class="MuiStack-root css-1v111ru-MuiStack-root"
      >
        <div
          class="icons row-actions"
          draggable="true"
          style="padding: 0px;"
        >
          <div
            class="charm-drag-handle disable-drag-selection MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-d1t9d3-MuiSvgIcon-root"
              data-testid="DragIndicatorIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2m-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell title-cell MuiBox-root css-v7v99c"
      >
        <div
          style="display: flex; width: 100%;"
        >
          <div
            class="octo-icontitle"
            style="align-self: flex-start; align-items: flex-start;"
          >
            <div
              class="css-jm1ma0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hvmj5c-MuiSvgIcon-root"
                data-testid="InsertDriveFileOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zM6 20V4h7v5h5v11z"
                />
              </svg>
            </div>
            <div
              style="width: 100%; height: 100%;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-2umbt5-MuiInputBase-root"
                  spellcheck="true"
                >
                  <input
                    class="MuiInputBase-input Editable octo-propertyvalue css-yz9k0d-MuiInputBase-input"
                    placeholder="Untitled"
                    type="text"
                    value="Page Title"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="open-button"
            data-test="database-open-button-card-1"
          >
            <button
              class="Button size--small"
              type="button"
            >
              <span>
                Open
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="flex-grow: 1;"
      />
    </div>
  </span>
</div>
`;

exports[`components/table/TableRow should match snapshot, read-only 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="TableRow octo-table-row"
      data-test="database-row-card-1"
      style="background-color: transparent; opacity: 1; cursor: default;"
    >
      <div
        class="octo-table-cell title-cell MuiBox-root css-v7v99c"
      >
        <div
          style="display: flex; width: 100%;"
        >
          <div
            class="octo-icontitle"
            style="align-self: flex-start; align-items: flex-start;"
          >
            <div
              class="css-jm1ma0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hvmj5c-MuiSvgIcon-root"
                data-testid="InsertDriveFileOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zM6 20V4h7v5h5v11z"
                />
              </svg>
            </div>
            <div
              style="width: 100%; height: 100%;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-readOnly MuiInputBase-readOnly css-2umbt5-MuiInputBase-root"
                  spellcheck="true"
                >
                  <input
                    class="MuiInputBase-input Mui-readOnly Editable readonly octo-propertyvalue MuiInputBase-readOnly css-yz9k0d-MuiInputBase-input"
                    placeholder="Untitled"
                    readonly=""
                    type="text"
                    value="Page Title"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="open-button"
            data-test="database-open-button-card-1"
          >
            <button
              class="Button size--small"
              type="button"
            >
              <span>
                Open
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="flex-grow: 1;"
      />
    </div>
  </span>
</div>
`;

exports[`components/table/TableRow should match snapshot, resizing column 1`] = `
<div>
  <span
    class="variable variable"
  >
    <div
      class="TableRow octo-table-row hidden"
      data-test="database-row-card-1"
      style="background-color: transparent; opacity: 1; cursor: move;"
    >
      <div
        class="MuiStack-root css-1v111ru-MuiStack-root"
      >
        <div
          class="icons row-actions"
          draggable="true"
          style="padding: 0px;"
        >
          <div
            class="charm-drag-handle disable-drag-selection MuiBox-root css-0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-d1t9d3-MuiSvgIcon-root"
              data-testid="DragIndicatorIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2m-2-8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0-6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m6 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2m0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2m0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell title-cell MuiBox-root css-v7v99c"
      >
        <div
          style="display: flex; width: 100%;"
        >
          <div
            class="octo-icontitle"
            style="align-self: flex-start; align-items: flex-start;"
          >
            <div
              class="css-jm1ma0"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hvmj5c-MuiSvgIcon-root"
                data-testid="InsertDriveFileOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8zM6 20V4h7v5h5v11z"
                />
              </svg>
            </div>
            <div
              style="width: 100%; height: 100%;"
            >
              <div
                style="height: 100%;"
              >
                <div
                  class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-2umbt5-MuiInputBase-root"
                  spellcheck="true"
                >
                  <input
                    class="MuiInputBase-input Editable octo-propertyvalue css-yz9k0d-MuiInputBase-input"
                    placeholder="Untitled"
                    type="text"
                    value="Page Title"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="open-button"
            data-test="database-open-button-card-1"
          >
            <button
              class="Button size--small"
              type="button"
            >
              <span>
                Open
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="align-items: flex-start; width: 100px; overflow-x: hidden;"
      >
        <div
          style="width: 100%; height: 100%;"
        >
          <div
            style="height: 100%;"
          >
            <div
              class="MuiStack-root css-1gl94ry-MuiStack-root"
              data-test="closed-select-input"
            >
              <div
                class="MuiStack-root css-1a59yxd-MuiStack-root"
                data-test="select-preview"
              >
                <div
                  class="MuiStack-root css-b3ouwx-MuiStack-root"
                  data-mui-internal-clone-element="true"
                >
                  <div
                    class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorTurquoise MuiChip-filledTurquoise css-zuoxmo-MuiChip-root"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                    >
                      value 1
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="align-items: flex-start; width: 100px; overflow-x: hidden;"
      >
        <div
          style="width: 100%; height: 100%;"
        >
          <div
            style="height: 100%;"
          >
            <div
              class="MuiStack-root css-1gl94ry-MuiStack-root"
              data-test="closed-select-input"
            >
              <div
                class="MuiStack-root css-1a59yxd-MuiStack-root"
                data-test="select-preview"
              >
                <div
                  class="MuiStack-root css-b3ouwx-MuiStack-root"
                  data-mui-internal-clone-element="true"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="octo-table-cell"
        style="flex-grow: 1;"
      />
    </div>
  </span>
</div>
`;
